appengine/genetics/js/visualization.js

```diff
@@ -0,0 +1,563 @@
+  Genetics.Visualization.lastFrame = now;
+  Genetics.Visualization.lastDelay = delay;
+};
+
+/**
+ * Visualize the current state of the cage simulation (Genetics.Cage).
+ * @private
+ */
+Genetics.Visualization.display_ = function() {
+  // TODO(kozbial) draw current state of the screen.
+  Genetics.Visualization.populationChartWrapper_.draw();
+  Genetics.Visualization.pickFightChartWrapper_.draw();
+  Genetics.Visualization.chooseMateChartWrapper_.draw();
+  Genetics.Visualization.mateAnswerChartWrapper_.draw();
+};
+
+/**
+ * Count for events to keep track of time passed for charts.
+ * @type {number}
+ */
+Genetics.Visualization.eventNumber = 0;
+
+/**
+ * Returns a string representation of a mouse's properties
+ * @param {Genetics.Mouse} mouse The mouse to get information from.
+ * @return {String} A string representation of the mouse containing information
+ * about the mouse.
+ * @private
+ */
+Genetics.Visualization.getMouseInfo_ = function(mouse) {
+  return 'Mouse' + mouse.id + '(sex:' + mouse.sex + ' , size:' + mouse.size +
+      ' , aggressiveness:' + mouse.startAggressiveness + ' , fertility:' +
+      mouse.startFertility + ' , pickFight:' + mouse.pickFightOwner +
+      '/chooseMate:' + mouse.chooseMateOwner + '/mateAnswer:' +
+      mouse.mateAnswerOwner + ')';
+};
+
+/**
+ * Process events in Cage events queue.
+ * @private
+ */
+Genetics.Visualization.processCageEvents_ = function() {
+  // Handle any queued events.
+  var getMouseName = Genetics.Visualization.getMouseName;
+  while (Genetics.Cage.EVENTS.length) {
+    var event = Genetics.Cage.EVENTS.shift();
+    switch (event['TYPE']) {
+      case 'ADD':
+        var mouse = event['MOUSE'];
+        Genetics.Visualization.addMouse_(mouse);
+        console.log(getMouseName(mouse, true, true) + ' added to game.');
+        break;
+      case 'START_GAME':
+        Genetics.Visualization.updateChartData_();
+        break;
+      case 'FIGHT':
+          var instigatingMouse = Genetics.Visualization.MICE[event['ID']];
+        switch (event['RESULT']) {
+          case 'NONE':
+            console.log(getMouseName(instigatingMouse) +
+                ' elected to never fight again.');
+            break;
+          case 'INVALID':
+            console.log(getMouseName(instigatingMouse) +
+                ' is confused and wont fight again.');
+            break;
+          case 'SELF':
+            console.log(getMouseName(instigatingMouse) +
+                ' chose itself when asked whom to fight with. ' +
+                getMouseName(instigatingMouse) + ' is being executed to put ' +
+                'it out of its misery.');
+            Genetics.Visualization.removeMouse_(instigatingMouse);
+            break;
+          case 'WIN':
+            var opponent = Genetics.Visualization.MICE[event['OPT_OPPONENT']];
+            console.log(getMouseName(instigatingMouse) + ' fights and kills ' +
```

Comment:
Let's make a Genetics.log (that calls console.log), and call it instead.  That way we can redirect the log elsewhere in the future.